### PR TITLE
Add more undocumented Nop instructions required for nestest

### PR DESCRIPTION
**Pull request by johnor**

### CODE REVIEW

File: core/test/src/test_cpu.cpp [hunk 1]
```diff
@@ -2445,6 +2452,47 @@ TEST_F(CpuTest, nop04_zero) {
...
+    EXPECT_CALL(mmu, read_byte(registers.pc + 1)).WillOnce(Return(0xCD));
+    EXPECT_CALL(mmu, read_byte(registers.pc + 2)).WillOnce(Return(0xEF));
+
+    step_execution(4);
+    EXPECT_EQ(expected, registers);
+}
+class NopZeroXFixture : public CpuTest,
+                        public testing::WithParamInterface<uint8_t> {};
+
+TEST_P(NopZeroXFixture, nop14_zerox) {
```
Comment (robinlinden) [inline core/test/src/test_cpu.cpp]: "nop14_zerox"?

File: core/test/src/test_cpu.cpp [hunk 2]
```diff
@@ -2445,6 +2452,47 @@ TEST_F(CpuTest, nop04_zero) {
...
+INSTANTIATE_TEST_SUITE_P(NopZeroXadsfsadf,
+        NopZeroXFixture,
+        testing::Values(NOP_ZEROX14,
+                NOP_ZEROX34,
+                NOP_ZEROX54,
+                NOP_ZEROX74,
+                NOP_ZEROXD4,
+                NOP_ZEROXF4));
+
+TEST_F(CpuTest, nop34_zerox) {
```
Comment (robinlinden) [inline core/test/src/test_cpu.cpp]: "nop34_zerox"?

File: core/test/src/test_cpu.cpp [hunk 3]
```diff
@@ -2445,6 +2452,47 @@ TEST_F(CpuTest, nop04_zero) {
...
+    stage_instruction(GetParam());
+    expected.x = registers.x = 0x01;
+    expected.pc += 1;
+    EXPECT_CALL(mmu, read_byte(registers.pc + 1)).WillOnce(Return(0xCD));
+    EXPECT_CALL(mmu, read_byte(0xCD)).WillOnce(Return(0xFD)); // Dummy read
+
+    step_execution(4);
+    EXPECT_EQ(expected, registers);
+}
+INSTANTIATE_TEST_SUITE_P(NopZeroXadsfsadf,
```
Comment (robinlinden) [inline core/test/src/test_cpu.cpp]: adsfsadf? 😁

Comment (johnor) [reply core/test/src/test_cpu.cpp]: It is called like that :D

**Pull request merged**